src/mlpack/methods/reinforcement_learning/q_learning_impl.hpp

```diff
@@ -82,14 +125,16 @@ template <
   typename NetworkType,
   typename UpdaterType,
   typename BehaviorPolicyType,
-  typename ReplayType
+  typename ReplayType,
+  typename PrioritizedReplayType
 >
 double QLearning<
   EnvironmentType,
   NetworkType,
   UpdaterType,
   BehaviorPolicyType,
-  ReplayType
+  ReplayType,
```

Comment:
Maybe I missed something but can you use the ReplayType and PrioritizedReplayType at the same time?

Reply:
> Maybe I missed something but can you use the ReplayType and PrioritizedReplayType at the same time?

Of course, we can not two type of replay method simultaneously. But I didn't get what you say about this part. Can you be more specific or give an example?

Reply:
I thought we made all the changes to the interface to use experienced replay as the ReplayType.

Reply:
> I thought we made all the changes to the interface to use experienced replay as the ReplayType.

I am so sorry for my poor English :(. I still can not get the main idea. Do you mean we should keep the original template interface rather than add two kind of replay type  in here ?
```
template <
  typename EnvironmentType,
  typename NetworkType,
  typename UpdaterType,
  typename BehaviorPolicyType,
  typename ReplayType,
  typename PrioritizedReplayType
>
```

Reply:
No worries, keeping the current interface sounds like a possible solution to me, but maybe I missed an important part.

Reply:
> No worries, keeping the current interface sounds like a possible solution to me, but maybe I missed an important part.

Yes, I hope the discussion may help me to write the code more consistent with the codebase. I also learn many things from your comments. Thanks.